src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Serialization/ExtendableClientCapabilitiesJsonConverter.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using OmniSharp.Extensions.LanguageServer.Protocol.Client.Capabilities;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Serialization
+{
+    public class ExtendableClientCapabilitiesJsonConverter : JsonConverter
+    {
+        public static readonly ExtendableClientCapabilitiesJsonConverter Instance = new ExtendableClientCapabilitiesJsonConverter();
+
+        public override bool CanRead => true;
+
+        public override bool CanWrite => false;
+
+        public override bool CanConvert(Type objectType)
+        {
+            return typeof(ClientCapabilities).IsAssignableFrom(objectType);
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            var obj = JObject.Load(reader);
```

Comment:
`JObject` 🙈 

Realistically though, this should be a fairly small object, and it should only be used once during bootup. Additionally, O# uses it [extensively ](https://github.com/OmniSharp/csharp-language-server-protocol/tree/6371386ee90dde8a1c419f3dc8fc94e0c1bc35f0/src/Protocol/Serialization/Converters) in their serializers regardless.